File: mad3004-introd-swift/C0711561_F2017_MAD3004_FinalProject/C0711561_F2017_MAD3004_FinalProject/Motorcycle.swift

public class Motorcycle: Vehicle {
    
    private let vehicleType = "Motorcycle"
    private var cylinderCapacity: Int
    private var type: String //street, off-road, and dual purpose
    private var ridingPosture: String //sport, standard, cruiser
    
    override init() {
        cylinderCapacity = 0
        type = ""
        ridingPosture = ""
        super.init()
    }
    
    init(pMake: String, pModel: String, pPlate: String, pCylinderCapacity: Int, pType: String, pRidingPosture: String) {
        cylinderCapacity = pCylinderCapacity
        type = pType
        ridingPosture = pRidingPosture
        super.init(pMake: pMake, pModel: pModel, pPlate: pPlate)
    }
    
    override func getVehicleType() -> String {
        return vehicleType
    }
    
    override func printMyData() -> String {
        return "\(super.printMyData())\n  -Cylinder Capacity: \(cylinderCapacity)\n  -Type: \(type)\n  -Riding Posture: \(ridingPosture)\n"
    }
}

File: mad3464-programming-java-i/C0711561_F2017_MAD3464_FinalProject/src/c0711561/f2017/mad3464/finalproject/vehicle/Motorcycle.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package c0711561.f2017.mad3464.finalproject.vehicle;

/**
 *
 * @author macstudent
 */
public class Motorcycle extends Vehicle {
    
    private static final String VEHICLE_TYPE = "Motorcycle";
    
    private int cylinderCapacity;
    private String type; //street, off-road, racing, and dual purpose
    private String ridingPosture; //sport, standard, cruiser

    public Motorcycle(String make, String plate, int cylinderCapacity, String type, String ridingPosture) {
        super(make, plate);
        this.cylinderCapacity = cylinderCapacity;
        this.type = type;
        this.ridingPosture = ridingPosture;
    }

    @Override
    public String getVehicleType() {
        return VEHICLE_TYPE;
    }
}
